sort_by_key gives multiple definitions for the same mangled name

**Describe the Bug:**
Trying to compile
```C++
#include <sycl/sycl.hpp>

#include <oneapi/dpl/algorithm>
#include <oneapi/dpl/execution>

namespace Kokkos {
namespace Experimental {
namespace Impl {

template <class ValueType> class RandomAccessIterator {
public:
  using iterator_type = RandomAccessIterator<ValueType>;

  using iterator_category = std::random_access_iterator_tag;
  using value_type = ValueType;
  using difference_type = ptrdiff_t;
  using pointer = value_type *;
  using reference = value_type &;

  //  using is_passed_directly = std::true_type;

  RandomAccessIterator() = default;

  explicit RandomAccessIterator(value_type *data, int stride)
      : m_data(data), m_stride(stride) {}

  iterator_type &operator++() {
    if constexpr (is_always_contiguous)
      m_data++;
    else
      m_data += m_stride;
    return *this;
  }

  iterator_type operator++(int) {
    auto tmp = *this;
    ++*this;
    return tmp;
  }

  iterator_type &operator--() {
    if constexpr (is_always_contiguous)
      m_data--;
    else
      m_data -= m_stride;
    return *this;
  }

  iterator_type operator--(int) {
    auto tmp = *this;
    --*this;
    return tmp;
  }

  reference operator[](difference_type n) const {
    if constexpr (is_always_contiguous)
      return *(m_data + n);
    else
      return *(m_data + n * m_stride);
  }

  iterator_type &operator+=(difference_type n) {
    if constexpr (is_always_contiguous)
      m_data += n;
    else
      m_data += n * m_stride;
    return *this;
  }

  iterator_type &operator-=(difference_type n) {
    if constexpr (is_always_contiguous)
      m_data -= n;
    else
      m_data -= n * m_stride;
    return *this;
  }

  iterator_type operator+(difference_type n) const {
    auto it = *this;
    it += n;
    return it;
  }

  iterator_type operator-(difference_type n) const {
    auto it = *this;
    it -= n;
    return it;
  }

  difference_type operator-(iterator_type it) const {
    if constexpr (is_always_contiguous)
      return m_data - it.m_data;
    else
      return (m_data - it.m_data) / m_stride;
  }

  bool operator==(iterator_type other) const { return m_data == other.m_data; }

  bool operator!=(iterator_type other) const { return m_data != other.m_data; }

  bool operator<(iterator_type other) const { return m_data < other.m_data; }

  bool operator<=(iterator_type other) const { return m_data <= other.m_data; }

  bool operator>(iterator_type other) const { return m_data > other.m_data; }

  bool operator>=(iterator_type other) const { return m_data >= other.m_data; }

  reference operator*() const { return *m_data; }

  pointer data() const { return m_data; }

  int stride() const { return m_stride; }

  friend iterator_type operator+(difference_type n, iterator_type other) {
    return other + n;
  }

private:
  pointer m_data;
  int m_stride;
  static constexpr bool is_always_contiguous = false;

  // Needed for the converting constructor accepting another iterator
  template <class> friend class RandomAccessIterator;
};

} // namespace Impl
} // namespace Experimental
} // namespace Kokkos

template <class T>
struct sycl::is_device_copyable<
    Kokkos::Experimental::Impl::RandomAccessIterator<T>> : std::true_type {};

int main() {
  sycl::queue queue;
  auto policy = oneapi::dpl::execution::make_device_policy(queue);

  oneapi::dpl::sort_by_key(policy,
                           ::Kokkos::Experimental::Impl::RandomAccessIterator(
                               static_cast<int *>(nullptr), 1),
                           ::Kokkos::Experimental::Impl::RandomAccessIterator(
                               static_cast<int *>(nullptr), 1),
                           ::Kokkos::Experimental::Impl::RandomAccessIterator(
                               static_cast<int *>(nullptr), 1));

  oneapi::dpl::sort_by_key(policy, static_cast<int *>(nullptr),
                           static_cast<int *>(nullptr),
                           static_cast<int *>(nullptr));
}
```
via 
```bash
icpx -fsycl ~/sort_by_key_bug_formatted.cpp 
```
gives
```bash
In file included from /home/darndt/sort_by_key_bug_formatted.cpp:4:
In file included from /soft/preview/pe/24.180.0-RC5/oneapi/dpl/latest/include/oneapi/dpl/execution:34:
In file included from /soft/preview/pe/24.180.0-RC5/oneapi/dpl/latest/include/oneapi/dpl/pstl/glue_algorithm_impl.h:26:
In file included from /soft/preview/pe/24.180.0-RC5/oneapi/dpl/latest/include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h:22:
In file included from /soft/preview/pe/24.180.0-RC5/oneapi/dpl/latest/include/oneapi/dpl/pstl/hetero/../parallel_backend.h:32:
In file included from /soft/preview/pe/24.180.0-RC5/oneapi/dpl/latest/include/oneapi/dpl/pstl/hetero/../hetero/dpcpp/parallel_backend_sycl.h:45:
/soft/preview/pe/24.180.0-RC5/oneapi/dpl/latest/include/oneapi/dpl/pstl/hetero/../hetero/dpcpp/parallel_backend_sycl_radix_sort.h:207:67: error: definition with same mangled name '_ZTSN6oneapi3dpl20__par_backend_hetero10__internal11__compositeINS1_25__radix_sort_count_kernelILj4ELb1ELb0EJEEEJLc95ELc90ELc84ELc83ELc78ELc54ELc111ELc110ELc101ELc97ELc112ELc105ELc51ELc100ELc112ELc108ELc50ELc48ELc95ELc95ELc112ELc97ELc114ELc95ELc98ELc97ELc99ELc107ELc101ELc110ELc100ELc95ELc104ELc101ELc116ELc101ELc114ELc111ELc50ELc53ELc95ELc95ELc114ELc97ELc100ELc105ELc120ELc95ELc115ELc111ELc114ELc116ELc95ELc99ELc111ELc117ELc110ELc116ELc95ELc107ELc101ELc114ELc110ELc101ELc108ELc73ELc76ELc106ELc52ELc69ELc76ELc98ELc49ELc69ELc76ELc98ELc48ELc69ELc74ELc78ELc83ELc48ELc95ELc57ELc101ELc120ELc101ELc99ELc117ELc116ELc105ELc111ELc110ELc53ELc95ELc95ELc100ELc112ELc108ELc49ELc55ELc68ELc101ELc102ELc97ELc117ELc108ELc116ELc75ELc101ELc114ELc110ELc101ELc108ELc78ELc97ELc109ELc101ELc69ELc82ELc78ELc83ELc52ELc95ELc49ELc51ELc100ELc101ELc118ELc105ELc99ELc101ELc95ELc112ELc111ELc108ELc105ELc99ELc121ELc73ELc83ELc53ELc95ELc69ELc69ELc78ELc83ELc48ELc95ELc56ELc95ELc95ELc114ELc97ELc110ELc103ELc101ELc115ELc56ELc97ELc108ELc108ELc95ELc118ELc105ELc101ELc119ELc73ELc78ELc83ELc48ELc95ELc49ELc48ELc95ELc95ELc105ELc110ELc116ELc101ELc114ELc110ELc97ELc108ELc53ELc116ELc117ELc112ELc108ELc101ELc73ELc74ELc105ELc105ELc69ELc69ELc69ELc76ELc78ELc52ELc115ELc121ELc99ELc108ELc51ELc95ELc86ELc49ELc54ELc97ELc99ELc99ELc101ELc115ELc115ELc52ELc109ELc111ELc100ELc101ELc69ELc49ELc48ELc50ELc54ELc69ELc76ELc78ELc83ELc71ELc95ELc54ELc116ELc97ELc114ELc103ELc101ELc116ELc69ELc50ELc48ELc49ELc52ELc69ELc76ELc78ELc83ELc71ELc95ELc49ELc49ELc112ELc108ELc97ELc99ELc101ELc104ELc111ELc108ELc100ELc101ELc114ELc69ELc49ELc69ELc69ELc69ELc78ELc83ELc70ELc95ELc54ELc98ELc117ELc102ELc102ELc101ELc114ELc73ELc106ELc76ELc105ELc49ELc69ELc78ELc83ELc70ELc95ELc54ELc100ELc101ELc116ELc97ELc105ELc108ELc49ELc55ELc97ELc108ELc105ELc103ELc110ELc101ELc100ELc95ELc97ELc108ELc108ELc111ELc99ELc97ELc116ELc111ELc114ELc73ELc106ELc69ELc69ELc118ELc69ELc69ELc69ELc69ELc69EEEE' as another definition
  207 |             sycl::nd_range<1>(__segments * __wg_size, __wg_size), [=](sycl::nd_item<1> __self_item) {
      |                                                                   ^
/soft/preview/pe/24.180.0-RC5/oneapi/dpl/latest/include/oneapi/dpl/pstl/hetero/../hetero/dpcpp/parallel_backend_sycl_radix_sort.h:207:67: note: previous definition is here
1 error generated.
```

**To Reproduce:**
see above

The following information might be useful: 
 - CMake command: doesn't apply
 - oneDPL version: N/A
 - Compiler version: Intel(R) oneAPI DPC++/C++ Compiler 2025.0.0 (2025.x.0.20240629)
 - OS: SUSE Linux Enterprise Server 15 SP4	
 - Device and driver version: Intel(R) Data Center GPU Max 1550 1.3 [1.3.28202]

**Expected Behavior:**
The file should compile.

**Additional Context:**
N/A

--- a/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h
+++ b/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h
@@ -655,7 +655,7 @@ struct __parallel_radix_sort_iteration
         using _CustomName = oneapi::dpl::__internal::__policy_kernel_name<_ExecutionPolicy>;
         using _RadixCountKernel =
             __internal::__kernel_name_generator<__count_phase, _CustomName, _ExecutionPolicy, ::std::decay_t<_InRange>,
-                                                ::std::decay_t<_TmpBuf>>;
+                                                ::std::decay_t<_TmpBuf>, _Proj>;
         using _RadixLocalScanKernel = __internal::__kernel_name_generator<__local_scan_phase, _CustomName,
                                                                           _ExecutionPolicy, ::std::decay_t<_TmpBuf>>;
         using _RadixReorderPeerKernel =

--- a/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h
+++ b/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h
@@ -660,10 +660,10 @@ struct __parallel_radix_sort_iteration
                                                                           _ExecutionPolicy, ::std::decay_t<_TmpBuf>>;
         using _RadixReorderPeerKernel =
             __internal::__kernel_name_generator<__reorder_peer_phase, _CustomName, _ExecutionPolicy,
-                                                ::std::decay_t<_InRange>, ::std::decay_t<_OutRange>>;
+                                                ::std::decay_t<_InRange>, ::std::decay_t<_OutRange>, _Proj>;
         using _RadixReorderKernel =
             __internal::__kernel_name_generator<__reorder_phase, _CustomName, _ExecutionPolicy,
-                                                ::std::decay_t<_InRange>, ::std::decay_t<_OutRange>>;
+                                                ::std::decay_t<_InRange>, ::std::decay_t<_OutRange>, _Proj>;
 
         ::std::size_t __max_sg_size = oneapi::dpl::__internal::__max_sub_group_size(__exec);
         ::std::size_t __reorder_sg_size = __max_sg_size;

